Problem1/prob1.py

```diff
@@ -0,0 +1,24 @@
+def diff_of_lists(lis1,lis2):
+    if lis1 is None or lis2 is None:
+        raise ValueError("Input lists cannot be None")
+    
+    temp_combined = list(lis1) + list(lis2)
+    if not all(isinstance(i, int) for i in temp_combined):
+        raise TypeError("Only integers are allowed in the lists")
+    
+    if not isinstance(lis1, list):
+        lis1 = list(lis1)
+
+    if not isinstance(lis2, list):
+        lis2 = list(lis2)
+    
+    result = []
+
+    result = [i for i in lis1 if i not in lis2]
```

Comment:
Searching in sets are faster than searching in lists, set ~ O(1) while list ~ O(n)
```suggestion
    lis1 = list(lis1)
    lis2 = set(lis2)
    if (
        any(not isinstance(i, int) for i in lis1)
        or any(not isinstance(i, int) for i in lis2)
    ):
        raise TypeError("Only integers are allowed in list2")

    result = [i for i in lis1 if i not in lis2]
```